### PR TITLE
Add katal, gray to derived units

### PR DESCRIPTION
**Pull request by say4n**
Added `katal`, the SI unit of catalytic activity and `gray`, the SI unit of absorbed dose.

### CODE REVIEW

File: sympy/physics/units/definitions.py [hunk 1]
```diff
@@ -61,6 +61,8 @@
 
 optical_power = dioptre = D = Quantity("dioptre", 1/length, 1/meter)
 lux = lx = Quantity("lux", luminous_intensity/length**2, steradian*candela/meter**2)
+katal = kat = Quantity("katal", amount_of_substance/time, mol/second)
```
Comment (gxyd) [inline sympy/physics/units/definitions.py]: Can you write a comment for `gray` saying "Absorbed dose of ionizing radiation" similarly for `katal`

**Comment by gxyd** [comment]
Otherwise this looks good to me.

**Comment by gxyd** [comment]
I just saw this issue https://github.com/sympy/sympy/issues/13379 . Probably we should wait before, and may be document these units as well.

File: sympy/physics/units/definitions.py (hunk 1 above)
Comment (Sayan98) [reply sympy/physics/units/definitions.py]: Ok, I will be on it ASAP.

**Comment by schymans** [comment]
I think #13379 should be dealt with separately. This one is probably good to go, thanks for makeing si.py more complete!

**Review by schymans — APPROVED** [review]
Looks great, thanks!

**Review by gxyd — APPROVED** [review]
LGTM

**Comment by gxyd** [comment]
Thanks for your contribution.

**Pull request merged**